MAT-6288 Facility Locations component

adongare opened the pull request:
## MADiE PR

Jira Ticket: [MAT-6288](https://jira.cms.gov/browse/MAT-6288)
(Optional) Related Tickets:

### Summary

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

adongare committed:
MAT-6288 Facility Locations component

adongare committed:
MAT-6288 unit tests

adongare committed:
Merge branch 'develop' into MAT-6288

adongare committed:
MAT-6288 revert the change, got removed accidently

adongare committed:
MAT-6288 fix lint warning

adongare committed:
MAT-6288 remove default value(as default is false/undefined)

adongare committed:
MAT-6288 lint fixes

adongare committed:
MAT-6288 remove debug statements

RohitKandimalla commented on src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx:
```diff
@@ -262,6 +263,14 @@ const DisplayAttributeInputs = ({
+      case "FacilityLocation":
+        return (
+          <FacilityLocation
+            canEdit={true}
```
Recently we completed the work on handling Edit privileges for Editing QDM testcases. Even though all inputs in this  DisplayAttributeInputs has a canEdit field, it is not required to pass in when using it from DisplayAttributeInputs, especially if we had to hard code it. 

May be we can remove it, make it optional in "FacilityLocation". Looks like we just don't need to worry about canEdit in DisplayAttributeInputs.tsx

RohitKandimalla replied:
I just saw that canEdit is chained as a required prop for its children Components. Damn it !!  I guess we can leave it for now then.

adongare replied:
I'm not sure what was our plan with canEdit. all of the components used here have hardcoded canedit made me think that we are going to address in some other story. if that's not required then I'll just remove it. thanks for pointing it.

adongare replied:
yeah, it is chained. I made it optional with default true.

adongare committed:
MAT-6288 canEdit not required at all the levels.We can make it optional and default it to true.
this works because editing is restricted at data element table level and data element category menu level

adongare committed:
Merge branch 'develop' into MAT-6288

adongare committed:
Merge branch 'develop' into MAT-6288
